File: backend/app/schemas/quick_note.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class QuickNoteCreate(BaseModel):
    content: str = Field(..., min_length=1, max_length=10000)


class QuickNoteResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    content: str
    created_at: datetime
    updated_at: datetime


class QuickNoteList(BaseModel):
    notes: list[QuickNoteResponse]
    total: int


class QuickNoteImageUploadResponse(BaseModel):
    url: str


class QuickNoteBatchDelete(BaseModel):
    ids: list[UUID] = Field(..., min_length=1, max_length=200)


class QuickNoteBatchDeleteResponse(BaseModel):
    deleted: int


class QuickNoteBatchMerge(BaseModel):
    ids: list[UUID] = Field(..., min_length=2, max_length=200)


class QuickNoteBatchMergeResponse(BaseModel):
    note: QuickNoteResponse
    merged: int

File: frontend/src/types/quickNote.ts
export interface QuickNote {
  id: string;
  user_id: string;
  content: string;
  created_at: string;
  updated_at: string;
}

export interface QuickNoteList {
  notes: QuickNote[];
  total: number;
}

export interface QuickNoteListFilters {
  q?: string;
  dateFrom?: string;
  dateTo?: string;
}

export interface QuickNoteCreate {
  content: string;
}

export interface QuickNoteImageUploadResponse {
  url: string;
}

export interface QuickNoteBatchDeleteResponse {
  deleted: number;
}
